engine/events/magikarp.asm

```diff
@@ -108,36 +108,18 @@ CalcMagikarpLength:
 ;   de: wEnemyMonDVs
 ;   bc: wPlayerID
 
-; This function is poorly commented.
-
-; In short, it generates a value between 190 and 1786 using
-; a Magikarp's DVs and its trainer ID. This value is further
-; filtered in LoadEnemyMon to make longer Magikarp even rarer.
+; It generates a value between 190 and 1755 using a Magikarp's DVs
+; and its trainer ID. This value is further filtered in LoadEnemyMon
+; to make longer Magikarp even rarer.
 
 ; The value is generated from a lookup table.
-; The index is determined by the dv xored with the player's trainer id.
-
-; bc = rrc(dv[0]) ++ rrc(dv[1]) ^ rrc(id)
+; The index is determined by the DV xored with the player's trainer ID
+; and then stored in bc.
```

Comment:
```suggestion
; The index is determined by the DV xored with the Magikarp's Trainer ID
; and then stored in bc.
```
The player's Trainer ID is only used when called from `LoadEnemyMon`. When this function is called from `CheckMagikarpLength`, it uses Magikarp's Original Trainer's Trainer ID, not the player's. Even though the Magikarp technically doesn't have a Trainer ID when it is generated in the wild, I think it's clear enough what is being referred to if this is just generalized as "the Magikarp's Trainer ID".

Reply:
In this case it can feel misleading if we don't also clarify it uses the Player's ID when fighting against a wild Magikarp.

Reply:
That's completely reasonable. Perhaps something like this would be better
```suggestion
; The index is determined by the DV xored with a Trainer ID 
; (the player's for wild Magikarp; the OT's when checking owned Magikarp)
; then stored in bc.
```

Reply:
It sounds better.